test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js

```diff
@@ -245,6 +263,98 @@ describe('lib/plugins/aws/package/lib/mergeIamTemplates.test.js', () => {
       });
     });
 
+    describe('Provider properties', () => {
+      let cfResources;
+      let naming;
+
+      before(async () => {
+        const { cfTemplate, awsNaming } = await runServerless({
+          fixture: 'function',
+          cliArgs: ['package'],
+          configExt: {
+            provider: {
+              iam: {
+                role: {
+                  statements: [
+                    {
+                      Effect: 'Allow',
+                      Resource: '*',
+                      NotAction: 'iam:DeleteUser',
+                    },
+                  ],
+                  managedPolicies: [
+                    'arn:aws:iam::123456789012:user/*',
+                    'arn:aws:s3:::my_corporate_bucket/Development/*',
+                    'arn:aws:iam::123456789012:u*',
+                  ],
+                  permissionBoundary: ['arn:aws:iam::123456789012:policy/XCompanyBoundaries'],
+                },
+              },
+              vpc: {
+                securityGroupIds: ['xxx'],
+                subnetIds: ['xxx'],
+              },
+              logRetentionInDays: 5,
+            },
+          },
+        });
+
+        cfResources = cfTemplate.Resources;
+        naming = awsNaming;
+      });
+
+      it('should support `provider.iam.role.statements`', async () => {
+        const IamRoleLambdaExecution = naming.getRoleLogicalId();
+        const iamResource = cfResources[IamRoleLambdaExecution];
+        const { Statement } = iamResource.Properties.Policies[0].PolicyDocument;
+
+        expect(Statement).to.deep.includes({
+          Effect: 'Allow',
+          Resource: '*',
+          NotAction: ['iam:DeleteUser'],
+        });
+      });
+
+      it('should support `provider.iam.role.managedPolicies`', () => {
+        const IamRoleLambdaExecution = naming.getRoleLogicalId();
+        const {
+          Properties: { ManagedPolicyArns },
+        } = cfResources[IamRoleLambdaExecution];
+
+        expect(ManagedPolicyArns).to.deep.includes('arn:aws:iam::123456789012:user/*');
+        expect(ManagedPolicyArns).to.deep.includes(
+          'arn:aws:s3:::my_corporate_bucket/Development/*'
+        );
+        expect(ManagedPolicyArns).to.deep.includes('arn:aws:iam::123456789012:u*');
+      });
+
+      it('should support `provider.iam.role.permissionsBoundary`', () => {
+        const IamRoleLambdaExecution = naming.getRoleLogicalId();
+        const {
+          Properties: { PermissionsBoundary },
+        } = cfResources[IamRoleLambdaExecution];
+        expect(PermissionsBoundary).to.be.equal(
+          'arn:aws:iam::123456789012:policy/XCompanyBoundaries'
+        );
+      });
+
+      it('should ensure needed IAM configuration when `provider.vpc` is configured', () => {
```

Comment:
Let's remove that test from `Provider properties - deprecated properties` block (at least to my understanding this is not influenced by properties we were handling here, so we're now testing twice same thing)